frontend/src/app/modules/time-series/services/line-chart.service.ts

```diff
@@ -245,6 +248,35 @@ export class LineChartService {
       .nice();
   }
 
+  private getLegendHeight(): number {
```

Comment:
Don't return a value to avoid side effects:
```suggestion
  private calculateLegendDimensions(): void {
```